src/NUnitTestAdapter/NUnit3TestExecutor.cs

```diff
@@ -266,7 +266,13 @@ private void RunAssembly(string assemblyName, TestFilter filter)
                     }
                 }
                 else
-                    TestLog.Info("NUnit failed to load " + assemblyName);
+                {
+                    var msgNode = loadResult.SelectSingleNode("properties/property[@name='_SKIPREASON']");
+                    if (msgNode != null && (new[] { "contains no tests", "Has no TestFixtures" }).Any(msgNode.GetAttribute("value").Contains))
+                        TestLog.Info("NUnit couldn't find any tests in " + assemblyName + ". (Probably an assembly dependency problem in your projects.");
```

Comment:
As a developer I understand your concerns, but as a user I would prefer some information instead of none. Some information will help some, no information will help none.

Your idea might work, but what about at tag before such "extra" information, and a disclaimer in the system documentation about what that tag implies? Our just a reference ID, and a documentation page collecting such information? (could even reference back to the commit/PR's where the issue was handled)